packages/react/src/popover/root/PopoverRoot.test.tsx

```diff
@@ -580,12 +580,18 @@ describe('<Popover.Root />', () => {
 
           await user.tab({ shift: true });
 
-          expect(screen.getByRole('button')).toHaveFocus();
+          await waitFor(() => {
+            expect(screen.getByRole('button')).toHaveFocus();
+          });
 
-          expect(screen.queryByTestId('popup')).to.toBeVisible();
+          await waitFor(() => {
+            expect(screen.queryByTestId('popup')).toBeVisible();
+          });
```

Comment:
Nitpick: this shouldn't be necessary. 🤔 
The above assertion already adds the "delay".

```suggestion
          expect(screen.queryByTestId('popup')).toBeVisible();
```

Reply:
How so? AFAIK `findByTestId` is async, not `queryByTestId`.